File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        stack<char> stk;
        string smp = "";
        for (char c : s){
            if ( c != '*' ){
                stk.push(c);
            }
            else{
                stk.pop();
            }
        }   
        while (not stk.empty() ){
            smp += stk.top(); 
            stk.pop();
        }
        // can also do while (!stk.empty()) ?
        /*
        for (int i = stk.size(); i > 0; i--){
            smp += stk.top(); 
            stk.pop();
        }
        */
        reverse(smp.begin(), smp.end()); 
        // testcase 3 is really long, cannot do stk.top() + smp;
        return smp;    
    }
};

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.py
class Solution(object):
    def removeStars(self, s):
        """
        :type s: str
        :rtype: str
        """
        stack = []
        string = ""
        for char in s:
            if char == '*' and len(stack) != 0:
                stack.pop()
            else:
                stack.append(char)
        for ch in stack:
            string += ch
        return string
        